java-agent/build.gradle.kts

```diff
@@ -145,7 +155,9 @@ kotlin {
                 from("src/native${it.compilationName.capitalize()}/kotlin")
                 into("src/${it.target.targetName}${it.compilationName.capitalize()}/kotlin/gen")
             }
-            it.compileKotlinTask.dependsOn(copyNativeClasses.get())
```

Comment:
Why? Possibly this will broke build process.

Reply:
compileKotlinTask was deprecated. Let me try to rewrite with compileTaskProvider

Reply:
reworked